packages/playground/build.ts

```diff
@@ -1,20 +1,249 @@
+        }), result);
+
+        Object.assign(result, combinedResults);
+
+        if (result.errors.length > 0) {
+          log("onEnd: errors in build, releasing lock on build directory");
+          lockfile.unlockSync(outdir);
+          return;
+        }
+
+        // If we have a any js files, that do not have a corresponding HTML file,
+        // we need to create one and embed the JavaScript into a <script> tag.
+        // Then we can delete the JavaScript file as it is no longer needed.
+        const outputs = result.metafile!.outputs;
+        for (const [jsPath, meta] of Object.entries(outputs)) {
+          if (jsExt.test(jsPath)) {
+            const htmlPath = jsPath.replace(/\.js$/, ".html");
+            if (!(htmlPath in outputs)) {
+              delete outputs[jsPath]
+              const js = await fsp.readFile(jsPath, { encoding: "utf8" });
+              const html = `<body></body><script>${js}</script>`;
+              await fsp.writeFile(htmlPath, html);
+              outputs[htmlPath] = { ...meta, bytes: meta.bytes + 30 };
+            }
+          }
+        }
+
+        // Use the user-provided renamer to generate a new name for the files, then
+        // update the filenames on disk, metafile.outputs and the importStubs (if present).
+        if (renamer) {
+          for (const [output, meta] of Object.entries(outputs)) {
+            const path = relative(outdir, output);
+            const newPath = renamer(path);
+            const newOutput = relative(__dirname, join(outdir, newPath));
+            log("Renaming:", path, "->", newPath);
+            await fsp.rename(output, newOutput);
+            if (path in importStubs) {
+              importStubs[newPath] = importStubs[path];
+            }
+            outputs[newOutput] = meta
+            delete outputs[output];
+            delete importStubs[path];
+          }
+        }
+
+        cleanupJS(outdir, log);
+
+        // Now we go through all of the output files and rewrite the import stubs to
+        // correct output path.
+        Object.keys(outputs)
+          .forEach(async output => {
+            let contents = await fsp.readFile(output, { encoding: "utf8" });
+            for (const [path, stub] of Object.entries(importStubs)) {
+              const replacement = pathPrefix + path;
+              log(`Replacing(${path}):`, stub, "->", replacement);
+              contents = contents.replaceAll(stub, replacement)
+            }
+            await fsp.writeFile(output, contents);
+          });
+
+
+        log("onEnd: releasing lock on build directory");
+        lockfile.unlockSync(outdir);
+
+        log("onEnd", JSON.stringify(result, undefined, 2));
+      });
+    },
+  })
+
+  return makePlugin(true);
+}
+
+const outdir = path.join(__dirname, "build")
+
 const buildOptions: esbuild.BuildOptions = {
-  entryPoints: {
-    index: "src/index.tsx",
-  },
+  entryPoints: ["./src/playground/index.tsx"],
+  outdir,
+  outbase: "src",
```

Comment:
We should force
```
treeShaking: true,
minify: true,
```
in the buildOptions (reduces the built duck example from 1MB to 143KB.